pkg/api/interfaces.go

```diff
@@ -48,10 +48,10 @@ type TimeInfo struct {
 
 // PodMetricsGetter knows how to fetch metrics for the containers in a pod.
 type PodMetricsGetter interface {
-	// GetContainerMetrics gets the latest metrics for all containers in each listed pod,
+	// GetPodMetrics gets the latest metrics for all containers in each listed pod,
 	// returning both the metrics and the associated collection timestamp.
 	// If a pod is missing, the container metrics should be nil for that pod.
-	GetContainerMetrics(pods ...apitypes.NamespacedName) ([]TimeInfo, [][]metrics.ContainerMetrics, error)
+	GetPodMetrics(pods ...apitypes.NamespacedName) ([]TimeInfo, [][]metrics.ContainerMetrics, error)
```

Comment:
Note that this is a breaking change for consumers of metrics-server as a library. For instance it will break prometheus-adapter which is implementing GetContainerMetrics. I am not against this change at all, I just wanted to mention that it will break some consumers so it might be worth mentioning that in the changelog.

Reply:
I am not sure how many projects are consuming metrics-server as a library, but would it make sense at some point to move part of the code to [custom-metrics-apiserver](https://github.com/kubernetes-sigs/custom-metrics-apiserver) in order to have all metrics API registration logic and interfaces in one place?

Reply:
That's true, but I don't consider using metrics-server as library as expected behavior. There is no benefits of using using MS code directly versus coping it. Until we clearly separate library code vs binary then I don't see a breaking change.

On the other hand, I had played with idea of keeping all metrics API machinery in one place. Basically merging metrics server `pkg/api` with https://github.com/kubernetes-sigs/custom-metrics-apiserver to create `metrics-api-machinery` that encompases both core, custom and external metrics. This way adapters that want to implement both custom and core api like Prometheus, can depend on one library instead of borrowing code from metrics-server.

Reply:
Ok, I wrote my response before I have seen yours.

Reply:
> That's true, but I don't consider using metrics-server as library as expected behavior.

Yes I completely agree, but I don't think prometheus-adapter is the only project that is doing that, hence my comment.

> I wrote my response before I have seen yours.

:exploding_head: looks like we have an agreement :smile:

Let's collaborate on that (if you want) once we have a bit more time on our hands. I already wanted to clean up part of the code for custom and external metrics and having a common interface for all metrics API will really help.
Having everything in one place will also help with the onboarding of new contributors.

That being said, would you suggest that we discuss the creation of `metrics-api-machinery` during the next SIG meeting?

Reply:
sg